### PR TITLE
Fixed exception in promise

### PR DESCRIPTION
**Pull request by TrySound**
Promise methods catches all exceptions and pass them to rejection

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -29,15 +29,11 @@ export default postcss.plugin('postcss-map', opts => {
...
-          maps[name] = yaml.safeLoad(data);
-          resolve();
-        } catch (e) {
-          reject(e);
-        }
+        resolve(data);
       });
+    }).then(function (data) {
+      let name = path.basename(map, path.extname(map));
+      maps[name] = yaml.safeLoad(data);
```
Comment (pascalduez) [inline lib/index.js]: So how do you catch exceptions from `yaml.safeload` now ?

Comment (TrySound) [inline lib/index.js]: As I said all methods implement catcing all expections inside them. Example of realization
https://github.com/floatdrop/pinkie/blob/master/index.js#L58-L63
Native do the same.

**Comment by TrySound** [comment]
fix #20

File: lib/index.js (hunk 1 above)
Comment (pascalduez) [inline lib/index.js]: Ok, I trust you judgement, don't have time to dwelve in it now.

**Pull request merged**